File: Coursera Courses/Specializations/Programming in Java: A Hands-on Introduction/Object-Oriented Java: Inheritance and Encapsulation/Week 2/LabChallenge.java
class MP3 {
  private String artist;
  private String title;
  private String album;
  private int length;
  private String genre;
  
  public String getArtist() {
    return artist;
  }
  
  public void setArtist(String newArtist) {
    artist = newArtist;
  }
  
  public String getTitle() {
    return title;
  }
  
  public void setTitle(String newTitle) {
    title = newTitle;
  }
  
  public String getAlbum() {
    return album;
  }
  
  public void setAlbum(String newAlbum) {
    album = newAlbum;
  }
  
  public int getLength() {
    return length;
  }
  
  public void setLength(int newLength) {
    length = newLength;
  }
  
  public String getGenre() {
    return genre;
  }
  
  public void setGenre(String newGenre) {
    genre = newGenre;
  }
}

//add class definitions below this line

class Podcast extends MP3 {
  private String name;
  private String date ;
  
  public Podcast(String n, String t, int l, String g, String d) {
    name = n;
    super.setTitle(t);
    super.setLength(l);
    super.setGenre(g);
    date = d;
  }
  
  public String displayName() {
    return "The name is "+name;
  }
  
  public String displayTitle() {
    return "The title is "+getTitle();
  }
  
  public String displayLength() {
    return "The length is "+getLength()/60+" minutes and "+getLength()%60+" seconds";
  }
  
  public String displayGenre() {
    return "The genre is "+getGenre();
  }
  
  public String displayDate() {
    return "The date is "+date;
  }
}

//add class definitions above this line

public class LabChallenge {
  public static void main(String[] args) {
    
    //add code below this line
  
    Podcast p = new Podcast("Planet Money", "Hollywood's Black List", 1460, "economics", "10 July 2020");
    System.out.println(p.displayName());
    System.out.println(p.displayTitle());
    System.out.println(p.displayLength());
    System.out.println(p.displayGenre());
    System.out.println(p.displayDate());
  
    //add code above this line   
  }
}

File: Coursera Courses/Specializations/Programming in Python: A Hands-on Introduction/Object-Oriented Python: Inheritance and Encapsulation/Week 1/lab_challenge.py
# parent class
class MP3:
  def __init__(self, artist, title, album, length, genre):
    self.artist = artist
    self.title = title
    self.album = album
    self.length = length
    self.genre = genre
    
  def get_artist(self):
    return f"The artist is {self.artist}"
    
  def get_title(self):
    return f"The title is {self.title}"
    
  def get_album(self):
    return f"The album is {self.album}"
    
  def get_length(self):
    return f"The length is {self.length}"
    
  def get_genre(self):
    return f"The genre is {self.genre}"
    
class Podcast(MP3):
  def __init__(self, name, title, length, genre, date): 
    self.name = name
    self.title = title
    self.length = length
    self.genre = genre
    self.date = date
    
  def get_name(self):
    return f"The name is {self.name}"
  
  def get_length(self):
    return f"The length is {self.length//60} minutes and {self.length%60} seconds"
  
  def get_date(self):
    return f"The date is {self.date}"
    
p = Podcast("Planet Money", "Hollywood's Black List", 1460, "economics", "10 July 2020")

print(p.get_name())
print(p.get_title())
print(p.get_length())
print(p.get_genre())
print(p.get_date())
